File: client/src/types/machine.ts
export interface Machine {
  id: number;
  siteId: number;
  name: string;
  type: string;
  serialNumber: string | null;
  purchaseDate: string | null;
  status: string;
}

export interface CreateMachineRequest {
  siteId: number;
  name: string;
  type: string;
  serialNumber: string;
  purchaseDate: string;
  status: string;
}

File: FieldOps.Domain/Entities/Machine.cs
namespace FieldOps.Domain.Entities;

public class Machine
{
    public int Id { get; set; }
    public int SiteId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string? SerialNumber { get; set; }
    public DateOnly? PurchaseDate { get; set; }
    public string Status { get; set; } = "active";
}
